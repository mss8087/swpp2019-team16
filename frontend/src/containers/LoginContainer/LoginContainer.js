import React, { useCallback } from 'react';
import { withRouter } from 'react-router-dom';
import Login from 'components/Login';
import { useDispatch } from 'react-redux';
import { login } from '../../modules/auth';

function LoginContainer({ history }) {
  const dispatch = useDispatch();

  const onLoginRequest = useCallback(
    ({ id, password }) => {
      dispatch(login({ id, password }));
    },
    [dispatch],
  );

  const onRegisterRequest = useCallback(() => {
    history.push('/signup'); //change to registerpage
  }, [history]);

  return (
    <Login
      onLoginRequest={onLoginRequest}
      onRegisterRequest={onRegisterRequest}
    />
  );
}
export default withRouter(LoginContainer);
