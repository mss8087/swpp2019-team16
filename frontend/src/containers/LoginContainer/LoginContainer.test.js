import React from 'react';
import LoginContainer from './LoginContainer';
import { renderWithRedux } from 'test/utils';
import { MemoryRouter, Route } from 'react-router-dom';

describe('<LoginContainer />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/login']}>
        <Route component={LoginContainer} path="/login" />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
