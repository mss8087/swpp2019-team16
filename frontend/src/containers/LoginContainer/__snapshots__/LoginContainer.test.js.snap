// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="loginPage"
  >
    <div
      class="title"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
        title="Login"
      >
        Login
      </h2>
    </div>
    <div
      class="inputID"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          ID
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-90 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-91"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="input"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-90 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-91"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      class="loginButton"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-92 makeStyles-root-95 MuiButton-contained makeStyles-contained-93 makeStyles-contained-96 MuiButton-contained makeStyles-contained-93 makeStyles-contained-96"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </span>
    <span>
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-92 makeStyles-root-129 MuiButton-contained makeStyles-contained-93 makeStyles-contained-130 MuiButton-contained makeStyles-contained-93 makeStyles-contained-130"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Register
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </span>
  </div>
</div>
`;
